src/internal_events/apache_metrics.rs

```diff
@@ -55,54 +54,3 @@ impl InternalEvent for ApacheMetricsParseError<'_> {
         );
     }
 }
-
-#[derive(Debug)]
-pub struct ApacheMetricsResponseError<'a> {
-    pub code: hyper::StatusCode,
-    pub endpoint: &'a str,
-}
-
-impl InternalEvent for ApacheMetricsResponseError<'_> {
-    fn emit(self) {
-        error!(
-            message = "HTTP error response.",
-            stage = error_stage::RECEIVING,
-            error_type = error_type::REQUEST_FAILED,
-            error_code = %http_error_code(self.code.as_u16()),
-            endpoint = %self.endpoint,
-            internal_log_rate_limit = true,
-        );
-        counter!(
-            "component_errors_total", 1,
-            "stage" => error_stage::RECEIVING,
-            "error_type" => error_type::REQUEST_FAILED,
-            "error_code" => http_error_code(self.code.as_u16()),
-            "endpoint" => self.endpoint.to_owned(),
```

Comment:
Side note, we might want to consider removing the `endpoint` (or `url`) tag from metrics and have the user rely on the logs for this information instead given the potential high cardinality.

Reply:
Ah, yes, agreed. I reopened and added it to https://github.com/vectordotdev/vector/issues/15426 to follow up on.